katz/src/main/kotlin/katz/predef.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2017 The Katz Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package katz
+
+/**
+ * Eagerly initialize all instances
+ **/
+//class Bootstrap() {
+//    companion object {
+//        init {
+//            println("called!")
+//            listOf(EitherMonad, IdMonad, NonEmptyListMonad, OptionMonad)
+//        }
+//    }
+//}
+
+object IntSemigroup : Semigroup<Int>, GlobalInstance<Semigroup<Int>>() {
```

Comment:
not in this PR, but we should create an issue to archive default Semigroup for language types Long, Float, String....